### PR TITLE
M2P-238 Unit Test: Bolt\Boltpay\Test\Unit\ThirdPartyModules\Mageplaza\ShippingRestrictionTest

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
 Unit Test: Bolt\Boltpay\Test\Unit\ThirdPartyModules\Mageplaza\ShippingRestrictionTest

Fixes: https://boltpay.atlassian.net/browse/M2P-238

#changelog M2P-238 Unit Test: Bolt\Boltpay\Test\Unit\ThirdPartyModules\Mageplaza\ShippingRestrictionTest

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

**Review by danac-gs — APPROVED** [review]

File: Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        );
+    }
+
+    /**
+     * @test
+     * @covers ::afterLoadSession
+     */
+    public function afterLoadSession()
+    {
+        $this->quote->expects(self::once())->method('getBoltParentQuoteId')->willReturn(self::QUOTE_ID);
```
Comment (vitaliyreznikov) [inline Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php:78]: I don't think we should have this type of tests.
In this test, we just check if **afterLoadSession** has all code it has now.
So if we need to refactor **afterLoadSession** we should rewrite the test.

In unit tests we should pass into method arguments, mock calls and check the result. In this case test should pass even after refactoring.

If we have no way to create a valuable test we shoudn't create tests at all.

As I mentioned in tech spec we need functional tests for all 3p plugins related code.

Let's wait for Matt's opinion here.

**Review by vitaliyreznikov — CHANGES_REQUESTED** [review]

File: Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php (hunk 1 above)
Comment (vitaliyreznikov) [reply Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php:78]: @matt-thomason

Comment (matt-thomason) [reply Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php:78]: @vitaliyreznikov @johnmccombs1 I agree here - I think tests to just bump coverage are not very useful.

Comment (danac-gs) [reply Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php:78]: > As I mentioned in tech spec we need functional tests for all 3p plugins related code.
How can we accomplish this?
I guess we need an instance per 3p module, with module deployed and configured, to run functional test against.

Comment (vitaliyreznikov) [reply Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php:78]: Ideally we need to do it in circle ci instance:
- install module
- configure module (such as create gift cart code)
- run tests
- delete module
- install next module

Hopefully, we can start this during Q4

**Pull request closed without merging**